docs/modules/ROOT/pages/access.adoc

```diff
@@ -133,6 +133,239 @@ there is no direct way for a contract to call circuits of other contracts or tra
+- `H(signature + context)` where `signature = sign(context)` - leverages wallet without exposing private key.
+
+*Context-Dependent Derivations:*
+
+- Include contract address, deployment timestamp, user ID, etc.
+- Trade-off: more context is more unique but harder to recreate.
+
+WARNING: Approaches that avoid private key exposure (public key + passphrase, signature-based)
+are generally recommended for operational security.
+
+Deriving the nonce deterministically from an <<air_gapped_public_key_agpk,Air-Gapped Public Key>> and user passphrase provides a balance of security and recoverability:
+
+```typescript
+// Example: Scrypt-based derivation
+import { scryptSync } from 'node:crypto';
+
+const deterministicNonce = scryptSync(
+  userPassphrase
+  publicKey + ":ZOwnablePK:nonce:v1",
+  32,
+  { N: 16384, r: 8, p: 1 } // Standard scrypt parameters
+);
+const recoverableOwnerId = ZOwnablePK._computeOwnerId(publicKey, deterministicNonce);
+```
+
+**Security Considerations**
+
+The `ZOwnablePK` module remains agnostic to nonce generation methods, placing the security/convenience decision entirely with the user. Key considerations include:
+
+- **Backup requirements**: Random nonces require additional secure storage.
+- **Recovery scenarios**: Deterministic nonces enable recovery.
+- **Cross-contract correlation**: Reusing nonce strategies may reduce privacy across deployments.
+- **Rotation costs**: Changing nonces requires ownership transfer transactions with associated DUST costs.
+
+Users should carefully evaluate their threat model, operational requirements,
+and privacy needs when selecting a nonce generation strategy,
+as this choice cannot be easily changed without transferring ownership.
+
+=== Air-Gapped Public Key (AGPK)
+
+For maximum privacy guarantees,
+users should employ an Air-Gapped Public Key (AGPK) exclusively for contract ownership and administrative circuits.
+An AGPK is a public key that maintains complete isolation from all other on-chain activities,
+similar to how air-gapped systems are isolated from networks to prevent data leakage.
+
+==== The Privacy Enhancement
+
+While `ZOwnablePK` provides cryptographic privacy through its commitment scheme,
+operational security practices like using an AGPK provide an additional layer of protection against correlation attacks. Even with the strongest cryptographic commitments,
+reusing a public key across different on-chain activities can potentially compromise privacy
+through transaction pattern analysis.
+
+==== AGPK Principles
+
+An Air-Gapped Public Key must adhere to strict isolation principles:
+
+- *Never used before:* The public key has no prior transaction history on any blockchain network.
+- *Never used elsewhere:* The key is exclusively reserved for the specific contract's administrative circuits i.e. `assertOnlyOwner`.
+- *Never used again:* The private key is destroyed once ownership is renounced or transferred to another account.
```

Comment:
I agree this is the best practice, but we should elaborate in the AGPK description that an AGPK is generated from a fresh private key pairing with no active public keys and that the expectation is the associated private key will be destroyed. That way users don't generate a new public key from a pre-existing private key with active public keys and later have to deal with the friction of migrating all funds to another fresh account or worse, simply continue using the account.

Reply:
Good call, on it!

Reply:
[418a7b5](https://github.com/OpenZeppelin/compact-contracts/pull/204/commits/418a7b564ae5276aef53beeb1a2dc452b0bd55b3) + [bd18c19](https://github.com/OpenZeppelin/compact-contracts/pull/204/commits/bd18c19094c59ed12748700d206f2a011539689a)